src/Atmos/Model/bc_energy.jl

```diff
@@ -203,3 +198,88 @@ function atmos_energy_normal_boundary_flux_second_order!(
     # NOTE: difference from design docs since normal points outwards
     fluxᵀn.energy.ρe -= C_h * ρ_avg * normu_int⁻_tan * (MSE - MSE_int)
 end
+
+"""
+    NishizawaEnergyFlux(fn) :: EnergyBC
+Calculate the net inward energy flux across the boundary following Nishizawa and Kitamura (2018).
+Return the flux (in W m^-2).
+"""
+struct NishizawaEnergyFlux{FNTM} <: EnergyBC
+    fn_T_and_q_tot::FNTM
+end
+function atmos_energy_boundary_state!(
+    nf,
+    bc_energy::NishizawaEnergyFlux,
+    atmos,
+    args...,
+) end
+function atmos_energy_normal_boundary_flux_second_order!(
+    nf,
+    bc_energy::NishizawaEnergyFlux,
+    atmos,
+    fluxᵀn,
+    n⁻,
+    state⁻,
+    diffusive⁻,
+    hyperdiffusive⁻,
+    aux⁻,
+    state⁺,
+    diffusive⁺,
+    hyperdiffusive⁺,
+    aux⁺,
+    t,
+    state_int⁻,
+    diffusive_int⁻,
+    aux_int⁻,
+)
+    FT = eltype(state⁻)
+    u_int⁻ = state_int⁻.ρu / state_int⁻.ρ
+    u_int⁻_tan = projection_tangential(atmos, aux_int⁻, u_int⁻)
+    normu_int⁻_tan = norm(u_int⁻_tan)
+    T, q_tot = bc_energy.fn_T_and_q_tot(atmos, state⁻, aux⁻, t)
+
+    q_tot_int =
+        atmos.moisture isa DryModel ? FT(0) :
+        state_int⁻.moisture.ρq_tot / state_int⁻.ρ
+
+    ## Initial guesses for MO parameters
+    LMO_init = eps(FT)
+    u_star_init = FT(0.1)
+    th_star_init = -FT(0.1)
+    qt_star_init = -FT(1e-5)
+    x_init = MArray{Tuple{4}, FT}(FT[
+        LMO_init,
+        u_star_init,
+        th_star_init,
+        qt_star_init,
+    ])
+    # Revisit this, these are not averaged values in DG
+    x_ave = MArray{Tuple{3}, FT}(FT[u_int⁻[1], aux_int⁻.moisture.θ_v, q_tot])
+    x_s = MArray{Tuple{3}, FT}(FT[state⁻.ρu[1] / state⁻.ρ, T, q_tot])
+    F_exchange = MArray{Tuple{3}, FT}(FT[0.01, 0.01, 0.01])
+    z_0 = MArray{Tuple{3}, FT}(FT[0.1, 0.1, 0.1])
+    Δz = altitude(atmos, aux_int⁻)
+    z = altitude(atmos, aux_int⁻)
+    θ_flux, q_tot_flux = get_energy_flux(
+        atmos.param_set,
+        x_init,
+        x_ave,
+        x_s,
+        z_0,
+        F_exchange,
+        aux_int⁻.moisture.θ_v,
+        q_tot_int,
+        Δz,
+        z,
+        DGScheme(),
+    )
+
+    # recover thermo state
+    ts_surf = PhaseEquil_ρTq(atmos.param_set, state⁻.ρ, T, q_tot)
+    # TODO: use the correct density at the surface
+    ρ_avg = average_density(state⁻.ρ, state_int⁻.ρ)
+    # Add sensible heat flux
+    fluxᵀn.energy.ρe -= ρ_avg * θ_flux * cp_m(ts_surf)
```

Comment:
I'm not sure whether `cp_m(ts_surf)` or `cp_m(ts_int)`, or the average of these two is better here. But the effect should be tiny.

Reply:
I agree. This is one of the things that I want to discuss... I think we might need to choose different values for FV or DG/CG

Reply:
I ended up using the `surf` values for all the terms in the flux. I think the flux is applied at `z=0`, so this is what makes sense to me.

Reply:
Following the discussion about where the momentum flux is defined as well, I think we need to have the guidelines for how to do this written somewhere.